### DISCUSSION TITLE
How can I configure (org) discussions via REST API

### DISCUSSION THREAD

**Discussion by uwehdaub** [community/community#50911] [Discussions] [answered]
### Select Topic Area

Question

### Body

I want to create lots of organizations in our Enterprise Account.

And I want them to have organization discussions enabled.
How can I do so via the REST API.

I don't want to it for every organization via the UI.

BR Uwe Daub

**Comment by AzizBenIsmail** [comment]
To enable organization discussions via REST API, you can use the GitHub API. Here are the steps 

- [ ] you can follow:

1. Authenticate with the GitHub API using an access token with the admin:org scope.

2. Get the ID of the organization for which you want to enable discussions using the /orgs/{org} endpoint.

3. Create a team in the organization that will have access to the discussions using the /orgs/{org}/teams endpoint.

4. Create a discussion category for the organization using the /orgs/{org}/teams/{team_slug}/discussion_categories endpoint.

5. Set the can_create_discussions parameter of the team to true using the /teams/{team_id} endpoint.

6. Set the discussion_category_id parameter of the team to the ID of the discussion category created in step 4 using the /teams/{team_id}/discussions endpoint.

7. Repeat steps 3 to 6 for each organization that you want to enable discussions for.

Here is an example of the curl command that you can use to enable organization discussions for a specific organization:


Copy 

- [ ] code

curl -X PATCH -H "Authorization: token YOUR_TOKEN" -H "Accept: application/vnd.github.scarlet-witch-preview+json" https://api.github.com/teams/TEAM_ID -d '{"discussion_category_id": DISCUSSION_CATEGORY_ID, "permission": "push", "can_create_discussions": true}'

- Replace YOUR_TOKEN with your personal access token, TEAM_ID with the ID of the team that you created, and DISCUSSION_CATEGORY_ID with the ID of the discussion category that you created.

- Note that the application/vnd.github.scarlet-witch-preview+json header is required to enable the Discussions feature preview in the GitHub API.

**Comment by jricciardi** [comment, marked as answer]
Hi @uwehdaub! There is not currently a way to enable org discussions through any API right now, as of March 24, 2023.
